packages/astro/src/build/util.ts

```diff
@@ -21,13 +21,21 @@ export function getDistPath(specifier: string, { astroConfig, srcPath }: { astro
 
   const fileLoc = new URL(specifier, srcPath);
   const projectLoc = fileLoc.pathname.replace(projectRoot.pathname, '');
+  const ext = path.extname(fileLoc.pathname);
 
-  const isPage = fileLoc.pathname.includes(pagesRoot.pathname);
-  // if this lives above src/pages, return that URL
+  const isPage = fileLoc.pathname.includes(pagesRoot.pathname) && (ext === '.astro' || ext === '.md');
```

Comment:
This is important: I considered it sacred that `src/pages` only contained Markdown and Astro files. non-pages shouldn't live in `src/pages`, right? `public/` is a much better place (or `src/assets` if for some reason they need to be inside of src).

Can we instead throw a warning/error if the extension doesn't match?

Reply:
This hasn’t been the case for a while, since we decided that there is no “Astro” folder; only that some folder (which the user can configure; `src/pages` by default) maps 1:1 to the routing structure. But since day 1 we’ve said that you can import a file from anywhere in your project. That means if you imported from `src/components/*`, **that has to have some final URL.**

That last part is important: it means that we really can’t restrict users from putting additional files within `src/pages`. Not unless we did more work to restrict that, and throw errors.

This PR doesn’t change fundamentally how Astro works at all; this is a behind-the-scenes change that only fixes bugs that existed before. This PR stops short of any opinions that affect Astro’s design. If we wanted to throw an error if a user put anything in `src/pages` other than `.astro` or `.md`, that would be a new change we’d have to discuss. And that would have to happen after this PR merged anyway since it’s additional work. This is a bugfix and nothing more.

Reply:
Ah, so If you build a site now and it has an image in `src/pages`, does the site build? Dev works as well?

If so, then I'm on board with treating this comment as unrelated / out of scope for this PR. Happy to spin off in a separate issue. If not lets hold off on merging a day or two so that we don't thrash the users on this.

Reply:
> Ah, so If you build a site now and it has an image in `src/pages`, does the site build? Dev works as well?

The problem wasn’t as apparent in dev as you had to try to load that file manually. But in build, we’d scan for files to make sure they all built, and that was where the error was. But yes, overall the TL;DR of this PR is “this fixes unexpected build failures in Astro just based off HTML you may have written”